pkg/chart/chart.go

```diff
@@ -78,31 +81,37 @@ type HTTPClient interface {
 	Do(req *http.Request) (*http.Response, error)
 }
 
-// LoadChart should return a Chart struct from an IOReader
-type LoadChart func(in io.Reader) (*chart.Chart, error)
+type ChartMultiVersion struct {
+	V2 *chartv2.Chart
+	V3 *chartv3.Chart
+}
+
+// LoadChartV2 should return a V2 Chart struct from an IOReader
+type LoadChartV2 func(in io.Reader) (*chartv2.Chart, error)
+
+// LoadChartV3 returns a V3 Chart struct from an IOReader
+type LoadChartV3 func(in io.Reader) (*chartv3.Chart, error)
 
 // Resolver for exposed funcs
 type Resolver interface {
 	ParseDetails(data []byte) (*Details, error)
-	GetChart(details *Details, netClient HTTPClient) (*chart.Chart, error)
+	GetChart(details *Details, netClient HTTPClient) (*ChartMultiVersion, error)
 	InitNetClient(details *Details) (HTTPClient, error)
 }
 
-// Chart struct contains the clients required to retrieve charts info
-type Chart struct {
+// ChartClient struct contains the clients required to retrieve charts info
+type ChartClient struct {
```

Comment:
I think I've mentioned this before, but I found it confusing that we had a struct called `Chart` which wasn't a chart, but a struct containing clients for getting chart details. Hence the rename.

Reply:
That's way less ambiguous